erpnext/accounts/doctype/payment_reconciliation/payment_reconciliation.py

```diff
@@ -24,15 +24,67 @@ def get_nonreconciled_payment_entries(self):
 			order_doctype, against_all_orders=True, against_account=self.bank_cash_account, limit=self.limit)
 		journal_entries = get_advance_journal_entries(self.party_type, self.party, self.receivable_payable_account,
 			order_doctype, against_all_orders=True, limit=self.limit)
+		dr_cr_notes_entries = self.get_dr_cr_notes_entries()
 				
-		self.add_payment_entries(payment_entries + journal_entries)
+		self.add_payment_entries(payment_entries + journal_entries + dr_cr_notes_entries)
 
 	def add_payment_entries(self, entries):
 		self.set('payments', [])
 		for e in entries:
 			row = self.append('payments', {})
 			row.update(e)
 
+	def get_dr_cr_notes_entries(self):
+		dr_cr_notes_entries = []
+
+		invoice_doctype = 'Sales Invoice' if self.party_type == 'Customer' else 'Purchase Invoice'
+
+		gl_entries = frappe.db.get_all(
+			"GL Entry",
+			filters={
+				"party_type": self.party_type,
+				"party": self.party,
+				"account": self.receivable_payable_account,
+				"voucher_type": invoice_doctype,
+			},
+			fields=["voucher_no"],
+			group_by="voucher_no"
+		)
+
+		if not gl_entries:
+			return []
+
+		invoice_names = [d.voucher_no for d in gl_entries]
+		if not invoice_names:
+			return []
+
+		dr_cr_invoices = frappe.get_all(
+			invoice_doctype,
+			filters={
+				'company': self.company,
+				'is_return': 1,
+				'docstatus': 1,
+				'name': ['in', invoice_names],
+				'outstanding_amount': ('<', 0)
+			},
+			fields=['name', 'posting_date', 'outstanding_amount', 'currency']
+		)
+
+		for inv in dr_cr_invoices:
+			dr_cr_notes_entries.append({
+				'reference_type': invoice_doctype,
+				'reference_name': inv.name,
+				'posting_date': inv.posting_date,
+				'reference_row': None,
+				'amount': abs(inv.outstanding_amount),
+				'allocated_amount': 0.0,
+				'difference_amount': 0.0,
+				'invoice_number': '',
+				'currency': inv.currency,
+			})
+
+		return dr_cr_notes_entries
+
```

Comment:
You can rewrite this method with reference from the same method in ERPNext standard.